packages/devtools_app/lib/src/charts/treemap.dart

```diff
@@ -0,0 +1,517 @@
+        final pRatio = pSize / (pSize + list2Size);
+
+        final pWidth =
+            isHorizontalRectangle ? pAndList2Ratio * width : pRatio * width;
+
+        final pHeight =
+            isHorizontalRectangle ? pRatio * height : pAndList2Ratio * height;
+
+        final pAspectRatio = pWidth / pHeight;
+
+        // Best aspect ratio that is the closest to 1.
+        if ((1 - pAspectRatio).abs() < (1 - pBestAspectRatio).abs()) {
+          pBestAspectRatio = pAspectRatio;
+          bestIndex = i;
+          // Kept track of width and height to construct the pivot cell.
+          pivotBestWidth = pWidth;
+          pivotBestHeight = pHeight;
+        }
+      }
+      // Split the rest of the data into list2 and list3
+      // [L1, pivotIndex, [L2 bestIndex], L3]
+      list2 = children.sublist(pivotIndex + 1, bestIndex + 1);
+      list2Size = computeByteSizeForNodes(
+        nodes: list2,
+        startIndex: 0,
+        endIndex: list2.length - 1,
+      );
+
+      list3 = children.sublist(bestIndex + 1);
+      list3Size = computeByteSizeForNodes(
+        nodes: list3,
+        startIndex: 0,
+        endIndex: list3.length - 1,
+      );
+    } else {
+      // Put all data in l2 and none in l3.
+      list2 = children.sublist(pivotIndex + 1);
+      list2Size = computeByteSizeForNodes(
+        nodes: list2,
+        startIndex: 0,
+        endIndex: list2.length - 1,
+      );
+
+      final pdAndList2Ratio = (pSize + list2Size) / totalSize;
+      final pdRatio = pSize / (pSize + list2Size);
+      pivotBestWidth =
+          isHorizontalRectangle ? pdAndList2Ratio * width : pdRatio * width;
+      pivotBestHeight =
+          isHorizontalRectangle ? pdRatio * height : pdAndList2Ratio * height;
+    }
+
+    // Contruct list 1 sub-treemap.
+    final list1SizeRatio = list1Size / totalSize;
+    final list1Width = isHorizontalRectangle ? width * list1SizeRatio : width;
+    final list1Height =
+        isHorizontalRectangle ? height : height * list1SizeRatio;
+    final list1Cells = buildTreemap(
+      children: list1,
+      width: list1Width,
+      height: list1Height,
+      x: x,
+      y: y,
+    );
+
+    // Construct list 2 sub-treemap.
+    final list2Width =
+        isHorizontalRectangle ? pivotBestWidth : width - pivotBestWidth;
+    final list2Height =
+        isHorizontalRectangle ? height - pivotBestHeight : pivotBestHeight;
+    final list2XCoord = isHorizontalRectangle ? x + list1Width : x;
+    final list2YCoord =
+        isHorizontalRectangle ? y + pivotBestHeight : y + list1Height;
+    final list2Cells = buildTreemap(
+      children: list2,
+      width: list2Width,
+      height: list2Height,
+      x: list2XCoord,
+      y: list2YCoord,
+    );
+
+    // Construct pivot cell.
+    final pivotXCoord = isHorizontalRectangle ? x + list1Width : x + list2Width;
+    final pivotYCorrd = isHorizontalRectangle ? y : y + list1Height;
+    final pivotCell = TreemapCell(
+      key: UniqueKey(),
```

Comment:
is this necessary?

Reply:
Without the key, flutter tries to reuse the widgets and the names of each cell does not update accordingly when zooming in